### PR TITLE
Add basic .travis.yml for testing on Travis-CI

### PR DESCRIPTION
**Pull request by spl**
This is a basic script for configuring, building, and testing on [Travis-CI](https://travis-ci.org/). In case you're not aware, Travis-CI provides free continuous-integration testing for open-source software. I've used it many times, and it can be very helpful to keep your code working. You can see this script run successfully [here](https://travis-ci.org/spl/teckit/builds/447656103).

To have this run, you will need to create a Travis-CI account, which you can do using your GitHub account. You will also need to flick a switch on the Travis-CI settings for this repository to enable it. The default setting is to run on every pull request and commit.

This script performs the simplest configuration, build, installation, and test run on Ubuntu Linux and macOS. There is plenty more you could extend this with in the future. I can imagine, for example, testing against multiple compiler versions and building install packages. Unfortunately, Travis-CI does not support Microsoft WIndows: there are other services for that, but I haven't used them.

### CODE REVIEW

**Comment by neilmayhew** [comment]
AppVeyor is often a good choice for Windows CI. Having CI with testing for all three supported platforms would be a big gain for TECkit.

**Comment by devosb** [comment]
I was already thinking of using Travis CI, as many of the other projects I track also use it. @neilmayhew 's comment made me realise that many of the same projects also use AppVeyor. I am not sure how useful AppVeyor's Windows support would be, as the TECkit Windows binaries are built on Linux using a cross-compiler. I will enable Travis and the configuration can be changed for out of tree builds, if needed AppVeyor can be added at a later point.

**Comment by neilmayhew** [comment]
I'd forgotten about cross builds being the default method for building Windows binary releases. However, it's easy to build TECkit on Windows using MinGW. The process is explained in detail in `INSTALL` and is almost identical to doing a cross-build on Linux, once you get the right tools installed. So it might not be hard to set it up in AppVeyor.

If not, it should at least be possible to have Travis produce Windows binaries via a cross-build and have AppVeyor pull those artifacts and run the tests.

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+matrix:
+  include:
+    - os: linux
+    - os: osx
+
+script:
+  - ./autogen.sh
+  - ./configure
+  - make
+  - sudo make install
```
Comment (neilmayhew) [inline .travis.yml:12]: It's not necessary to `make install` in order to run tests. A better version of the script would be:

```
./autogen.sh
mkdir build
cd build
make check
```

Detailed test output is saved to `test/dotests.pl.log`

Comment (spl) [reply .travis.yml:12]: I didn't put `make install` in for the tests but rather as a test of `make install` itself. Hopefully #12 suits this better version better. 🙂

**Pull request merged**